### ISSUE TITLE
AKS not exporting dns_prefix or vnet_subnet_id

### ISSUE THREAD

**Issue by cmeyertons** [terraform-providers/terraform-provider-azurerm#2371]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.11.9
+ provider.azurerm v1.19.0

### Affected Resource(s)

* azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
provider "azurerm" {
  version = "1.19"
}

variable "cluster_name" { }
variable "resource_group_name" { }

resource "azurerm_kubernetes_cluster" "aks" {
    name                = "${var.cluster_name}"
    location            = "${azurerm_resource_group.aks.location}"
    resource_group_name = "${azurerm_resource_group.aks.name}"
    dns_prefix          = "${var.dns_prefix}"
    kubernetes_version  = "1.10.8"

    linux_profile {
        admin_username = "ubuntu"

        ssh_key {
          key_data = "${file("${var.ssh_public_key}")}"
        }
    }

    agent_pool_profile {
        name            = "default"
        count           = "${var.agent_count}"
        vm_size         = "Standard_DS2_v2"
        os_type         = "Linux"
        os_disk_size_gb = 30
    }

    service_principal {
        client_id     = "${azurerm_azuread_application.aks.application_id}"
        client_secret = "${random_string.aks_sp_password.result}"
    }

    addon_profile {
      oms_agent {
        enabled                    = true
        log_analytics_workspace_id = "${var.log_analytics_workspace_id}"
      }
    }

    tags {
        Environment = "${var.environment}"
    }
}

data "azurerm_kubernetes_cluster" "aks-data" {
    name                = "${var.cluster_name}"
    resource_group_name = "${var.resource_group_name}"
}

output "agent_pool_profile" {
  # value = "${azurerm_kubernetes_cluster.aks.agent_pool_profile}"
  value = "${data.azurerm_kubernetes_cluster.aks-data.agent_pool_profile"
}
```

```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

agent_pool_profile = {
  count = 1
  dns_prefix =
  max_pods = 110
  name = default
  os_disk_size_gb = 30
  os_type = Linux
  vm_size = Standard_DS2_v2
  vnet_subnet_id =
}
```

### Expected Behavior

vnet_subnet_id & dns_prefix are exported in the `agent_pool_profile` using either data source or resource reference

### Actual Behavior

dns_prefix and vnet_subnet_id are both empty

### Steps to Reproduce

1. `terraform apply`

**Comment by tombuildsstuff** [comment]
hi @cmeyertons 

Thanks for opening this issue :)

Taking a quick look into this the `dns_prefix` field within the `agent_pool_profile` block is deprecated (whilst the AKS API exposed this field and thus we added support for it, it turns out it was never actually set-able/always returned an empty value and thus got removed after AKS launched) - instead there's a top-level field for `dns_prefix` which returns this information for the cluster.

Regarding the Virtual Network Subnet ID - since there's no Subnet configured for this resource (and there's no Network Profile specified in the Terraform Configuration above) - this value wouldn't be returned since it's not set. Whilst it may have been possible to configure this value through some other means (e.g. the Portal) - as far as I'm aware at this time the AKS API requires that the AKS Cluster is recreated when the Virtual Network Subnet ID is changed (e.g. added/removed). Out of interest, how did you assign the Subnet to the AKS Cluster?

Terraform runs a `refresh` automatically which queries the latest information in Azure during the Plan/Apply cycle - meaning these values should be up to date - that said it should be possible to verify this is the latest data being returned by running `terraform refresh` (to ensure the information in the state is updated with the latest info from Azure) and then `terraform output` (to output the values). Would you be able to take a look and see if that works for you? :)

Thanks!

**Comment by cmeyertons** [comment]
Makes sense on the `dns_prefix` - good to know.  Sounds like it was just a red herring as I saw two things not being returned

The one I really care about for my use case is the `vnet_subnet_id`.  My desired use case is:
1. Create AKS cluster w/ no specified networking (let azure do it for me)
2. Export the `vnet_subnet_id` that was generated
3. Leverage that `subnet_id` to set up an Azure DB VNET Firewall rule for an external Azure MySQL DB

I tried just hard coding the `subnet_id` manually from `resources.azure.com` and ended up getting an error stating that the subnet needs to be configured for the service endpoint `Microsoft.Sql/*` so I might be delving beyond the basic networking use case...
